### PR TITLE
chore: add docz

### PR DESCRIPTION
**Pull request by LucasRuy**
#### Pull Request Type:

- [ ] Bug :beetle:
- [x] Chore :gear:
- [ ] Feature :rocket:

#### Description of motivation for this Pull Request:

Added `docz` on this projects for create documentation easily.

### CODE REVIEW

File: docs/index.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+It is for [**React**](https://reactjs.org) apps and follows [**Standard Style Guide**](https://github.com/standard/standard).
+
+Fully focused library for developers and UX designers, always thinking about productivity gain.
+
+## Useful links
+
+To help you get started we leave below some links that may be useful for you:
+
+- [Getting Started](/getting-started) - Steps how to start using library.
+- [Documenting](/documenting) - Instructions for create documentation a new component.
```
Comment (leandrogrillo) [inline docs/index.mdx]:
```suggestion
- [Documenting](/documenting) - How to create a component documentation
```

nao sei.

File: docs/documenting/documentation.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+name: Documenting
+route: /documenting
+---
+
+# Documenting
+
+When creating a new component in this library, you need to create documentation for this component. If you can not, your _Pull Request_ will be disapproved by the team that will review your code.
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]:
```suggestion
When creating a new component in this library, you must document it. If you don't create the documentation for your component, your _Pull Request_ will be rejected by the reviewers.
```

File: docs/index.mdx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+- [Contribute](/contribute) - Instructions for people considering contribution to Backyard.
```
Comment (leandrogrillo) [inline docs/index.mdx]:
```suggestion
- [Contribute](/contribute) - How to contribute with Backyar.
```

File: docs/documenting/documentation.mdx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+Therefore, the following will be an example of how the documentation of the new components should be made to follow the pattern of development.
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]:
```suggestion
Therefore, the following is an example of how to make documentation of new components, following this library standards.
```

File: docs/documenting/documentation.mdx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+- [Starting](#starting)
+- [Structure of folders](#structure-of-folders)
+- [How to use](#how-to-use)
+- [Props](#props)
+
+## Introduction
+
+We encourage our developers to always create documentation of our components to facilitate the use of the user that will consume it.
+
+The purpose of this document is to show you how to proceed after you implement your component. The idea is that when you reach the end of this reading you will be able to follow our documentation standards.
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]:
```suggestion
The purpose of this document is to show to you how to proceed after you implement your component. The idea is that when you reach the end of this reading you will be able to follow our documentation standards.
```

File: docs/documenting/documentation.mdx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+## Starting
+
+Follow the steps below to begin building your component documentation.
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]: I don't think that we need this section, it is redundant.

File: docs/documenting/documentation.mdx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+## Structure of folders
+
+We have a mass structure for the components so we can follow them.
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]: I did not understand what this phrase means.

File: docs/documenting/documentation.mdx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+## Topics
+
+List of all topics in this document.
+
+- [Introduction](#introduction)
+- [Starting](#starting)
+- [Structure of folders](#structure-of-folders)
+- [How to use](#how-to-use)
+- [Props](#props)
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]: this link is broken

File: docs/documenting/documentation.mdx [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+You should create a folder named `docs` and inside it will be where the documentation of your component should be included. Simple components may contain only one file, more complex components may contain more than one file. But in cases of more extensive or complex components, we advise you to keep only one file in the folder and separate by topics to keep the rest of the documentation standard and also for organizational issues.
+
+In the end we expect its structure to be this way:
+
+![base](./images/base-structure-with-doc.png)
+
+## How to use
+
+We hope that the documentation for your component contains some topics to facilitate the use and understanding of the user that will consume this component.
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]: This phrase looks a lot like the one in introduction.

File: docs/documenting/documentation.mdx [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+Below the list of topics to be documented:
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]:
```suggestion
List of topics to be covered:
```

File: docs/index.mdx [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+Created based on Leroy Merlin Brazil needs, and used internally as well.
+
+It is for [**React**](https://reactjs.org) apps and follows [**Standard Style Guide**](https://github.com/standard/standard).
+
+Fully focused library for developers and UX designers, always thinking about productivity gain.
+
+## Useful links
+
+To help you get started we leave below some links that may be useful for you:
```
[the remark this answers is not in the document]
Comment (LucasRuy) [reply docs/index.mdx]: Vou procurar uma outra forma de formular essa frase então.

File: docs/documenting/documentation.mdx [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+- **How to use**
+  - This includes details of how the component should be called, in which cases it makes sense, and so on ...
+- **Variations**
+  - If your component has variations, documentation should contemplate these variations available.
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]:
```suggestion
  - If your component has variations, the documentation should contemplate them.
```

File: docs/documenting/documentation.mdx [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+- **Examples**
+  - This section should contain examples of how your component renders on the screen.
+- **Props**
+  - This section should be directed only to document the `API` available in your component. It should be created in tabular form with information about the available properties.
+
+## Example
```
Comment (ravanscafi) [inline docs/documenting/documentation.mdx]: I'm lost with this example.

File: docs/getting-started.mdx [hunk 14]
```diff
@@ -0,0 +1,45 @@
+---
+name: Getting Started
+route: /getting-started
+---
+
+# Getting Started
+
+## Prerequisites
+
+Prerequisites for install **Backyard UI**:
```
Comment (ravanscafi) [inline docs/getting-started.mdx]:
```suggestion
Prerequisites to install **Backyard UI**:
```

File: docs/getting-started.mdx [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+
+- [`Node.js`](https://nodejs.org/) version `>=8.0.0`.
```
Comment (ravanscafi) [inline docs/getting-started.mdx]: These prerequisites should be enforced on `package.json`:

```json
"engines": {
  "node": ">=8.0.0",
  "npm": ">=3.5.2",
  "yarn": "^1.16.0"
}
```

File: docs/getting-started.mdx [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+- [`Yarn`](https://yarnpkg.com/) version `1.16.0` or [`npm`](https://www.npmjs.com/get-npm) version `3.5.2`
+
+## Installation
+
+To install you needs runs this command on your terminal:
```
Comment (ravanscafi) [inline docs/getting-started.mdx]: you need
```suggestion
To install you need runs this command on your terminal:
```

File: docs/getting-started.mdx [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+
+Via `Yarn`:
+
+```
```
Comment (ravanscafi) [inline docs/getting-started.mdx]:
```suggestion
```bash
```

File: docs/getting-started.mdx [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+yarn install backyard-ui --dev
+```
+
+Via `npm`:
+
+```
```
Comment (ravanscafi) [inline docs/getting-started.mdx]:
```suggestion
```bash
```

File: docs/getting-started.mdx [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+npm install backyard-ui --save-dev
+```
+
+## Usage
+
+You can use components importing in your React application:
```
Comment (ravanscafi) [inline docs/getting-started.mdx]:
```suggestion
You can use components importing them in your React application:
```

File: docs/index.mdx [hunk 20]
```diff
@@ -0,0 +1,20 @@
+---
+name: Home
+route: /
+---
+
+# Backyard UI
+
+Created based on Leroy Merlin Brazil needs, and used internally as well.
+
+It is for [**React**](https://reactjs.org) apps and follows [**Standard Style Guide**](https://github.com/standard/standard).
```
Comment (ravanscafi) [inline docs/index.mdx]:
```suggestion
It is designed to work with [**React**](https://reactjs.org) apps and it follows the [**Standard Style Guide**](https://github.com/standard/standard).
```

File: docs/index.mdx (hunk 11 above)
Comment (ravanscafi) [inline docs/index.mdx]:
```suggestion
To learn more about **Backyard UI** you might want to check the resources below:
```

File: docs/index.mdx [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+
+Fully focused library for developers and UX designers, always thinking about productivity gain.
+
+## Useful links
+
+To help you get started we leave below some links that may be useful for you:
+
+- [Getting Started](/getting-started) - Steps how to start using library.
```
Comment (ravanscafi) [inline docs/index.mdx]:
```suggestion
- [Getting Started](/getting-started) - Steps on how to start using library.
```

File: docs/index.mdx (hunk 1 above)
Comment (ravanscafi) [inline docs/index.mdx]:
```suggestion
- [Documenting](/documenting) - Instructions to document a new component.
```

File: docs/index.mdx (hunk 3 above)
Comment (ravanscafi) [inline docs/index.mdx]:
```suggestion
- [Contribute](/contribute) - Instructions for people considering to contribute to Backyard UI.
```

File: docs/documenting/documentation.mdx (hunk 6 above)
Comment (LucasRuy) [reply docs/documenting/documentation.mdx]: Ok, i'm will remove this section from document.

File: docs/documenting/documentation.mdx (hunk 13 above)
Comment (LucasRuy) [reply docs/documenting/documentation.mdx]: i'm will remove this.

File: .gitignore [hunk 22]
```diff
@@ -1,2 +1,64 @@
-node_modules
-lerna-debug.log
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (leandrogrillo) [inline .gitignore]: none of these is needed here. `*.log` covers all `.log` files

Comment (leandrogrillo) [reply .gitignore]:
```
# Logs
logs
*.log
```

File: .gitignore [hunk 23]
```diff
@@ -1,2 +1,64 @@
...
+*.pid.lock
+
+# Directory for instrumented libs generated by jscoverage/JSCover
+lib-cov
+
+# Docz
+.docz/cache
+
+# Coverage directory used by tools like istanbul
+coverage
```
Comment (leandrogrillo) [inline .gitignore]: we don't need this in gitignore becase, we dont generate this folder

File: .gitignore [hunk 24]
```diff
@@ -1,2 +1,64 @@
...
+
+# nyc test coverage
+.nyc_output
```
Comment (leandrogrillo) [inline .gitignore]: we don't need this in gitignore becase, we dont generate this folder

File: .gitignore [hunk 25]
```diff
@@ -1,2 +1,64 @@
...
+
+# Grunt intermediate storage (http://gruntjs.com/creating-plugins#storing-task-files)
+.grunt
```
Comment (leandrogrillo) [inline .gitignore]: we don't need this in gitignore becase, we dont generate this file

File: .gitignore [hunk 26]
```diff
@@ -1,2 +1,64 @@
...
+
+# Bower dependency directory (https://bower.io/)
+bower_components
```
Comment (leandrogrillo) [inline .gitignore]: we don't need use bower, there is no need of this

File: .gitignore [hunk 27]
```diff
@@ -1,2 +1,64 @@
...
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
+
+# TypeScript v1 declaration files
+typings/
```
Comment (leandrogrillo) [inline .gitignore]: we don't need this in gitignore becase, we are not using `TS`

File: .gitignore [hunk 28]
```diff
@@ -1,2 +1,64 @@
...
+
+# Optional npm cache directory
+.npm
+
+# Optional eslint cache
+.eslintcache
+
+# Optional REPL history
+.node_repl_history
```
Comment (leandrogrillo) [inline .gitignore]: we don't need this in gitignore becase, we dont generate this folder

File: .gitignore [hunk 29]
```diff
@@ -1,2 +1,64 @@
...
+
+# Output of 'npm pack'
+*.tgz
+
+# Yarn Integrity file
+.yarn-integrity
```
Comment (leandrogrillo) [inline .gitignore]: we don't need this in gitignore becase, we dont generate this folder

File: .gitignore [hunk 30]
```diff
@@ -1,2 +1,64 @@
...
+
+# dotenv environment variables file
+.env
```
Comment (leandrogrillo) [inline .gitignore]: we don't need this in gitignore becase, we don't have .env file

File: .gitignore [hunk 31]
```diff
@@ -1,2 +1,64 @@
...
+
+# next.js build output
+.next
```
Comment (leandrogrillo) [inline .gitignore]: we don't need this in gitignore becase, we dont generate this file

File: .gitignore [hunk 32]
```diff
@@ -1,2 +1,35 @@
...
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
```
Comment (leandrogrillo) [inline .gitignore]: we dont need this rule

File: .gitignore [hunk 33]
```diff
@@ -1,2 +1,35 @@
...
+lib-cov
+
+# Docz
+.docz/
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
```
Comment (leandrogrillo) [inline .gitignore]: we dont need this rule

File: .gitignore [hunk 34]
```diff
@@ -1,2 +1,35 @@
...
+*.pid.lock
+
+# Directory for instrumented libs generated by jscoverage/JSCover
+lib-cov
+
+# Docz
+.docz/
+
+# node-waf configuration
+.lock-wscript
```
Comment (leandrogrillo) [inline .gitignore]: we dont need this rule

File: .gitignore [hunk 35]
```diff
@@ -1,2 +1,35 @@
...
+yarn-error.log*
+
+# Runtime data
+pids
+*.pid
+*.seed
+*.pid.lock
+
+# Directory for instrumented libs generated by jscoverage/JSCover
+lib-cov
```
Comment (leandrogrillo) [inline .gitignore]: we dont need this rule

File: .gitignore [hunk 36]
```diff
@@ -1,2 +1,35 @@
...
+# Logs
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# Runtime data
+pids
+*.pid
+*.seed
+*.pid.lock
```
Comment (leandrogrillo) [inline .gitignore]: we dont need these `pid` rules

**Pull request merged**